src/components/Api.js

```diff
@@ -0,0 +1,162 @@
+export default class Api {
+  constructor(apiSettings) {
+    this._apiSettings = apiSettings;
+    this._headers = apiSettings.headers;
+    this._baseUrl = apiSettings.baseUrl;
+    this._userInfoUrl = `${this._baseUrl}/users/me`;
+    this._cardsUrl = `${this._baseUrl}/cards`;
+    this._avatarUrl = `${this._userInfoUrl}/avatar`;
+  }
+
+  _checkResponse(response) {
+    if (!response.ok) {
+      throw response;
+    }
+  }
+
+  async getInitialCards() {
+    try {
+      const res = await fetch(this._cardsUrl, {
+        headers: this._headers,
+      });
+      this._checkResponse(res);
+      const cardsArray = await res.json();
+      console.log(cardsArray);
+      return cardsArray;
+    } catch (err) {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.

Please, remove it from `Api` and add it in `index.js`.


And not need to catch the error and throw it without changing. So, you 100% don't need it in `App`